cmd/crank/beta/convert/pipelinecomposition/converter.go

```diff
@@ -147,15 +147,98 @@ func processFunctionInput(input *Input) *runtime.RawExtension {
 		"apiVersion":  "pt.fn.crossplane.io/v1beta1",
 		"kind":        "Resources",
 		"environment": processedInput.Environment.DeepCopy(),
-		"patchSets":   processedInput.PatchSets,
-		"resources":   processedInput.Resources,
+		"patchSets":   MigratePatchPolicy(processedInput.PatchSets),
+		"resources":   MigratePatchPolicyInResources(processedInput.Resources),
 	}
 
 	return &runtime.RawExtension{
 		Object: &unstructured.Unstructured{Object: inputType},
 	}
 }
 
+// MigratePatchPolicyInResources patches resources.
+func MigratePatchPolicyInResources(resources []v1.ComposedTemplate) []ComposedTemplate {
+	composedTemplates := []ComposedTemplate{}
+
+	for _, resource := range resources {
+		composedTemplate := ComposedTemplate{}
+		composedTemplate.ComposedTemplate = resource
+		composedTemplate.Patches = patchPatches(resource.Patches)
+
+		composedTemplate.ComposedTemplate.Patches = nil
+		composedTemplates = append(composedTemplates, composedTemplate)
+	}
+	return composedTemplates
+}
+
+// MigratePatchPolicy migrates mergeoptions.
+func MigratePatchPolicy(patchset []v1.PatchSet) []PatchSet {
+	newPatchSets := []PatchSet{}
+
+	for _, patchSet := range patchset {
+		newpatchset := PatchSet{}
+		newpatchset.Name = patchSet.Name
+		newpatchset.Patch = patchPatches(patchSet.Patches)
+
+		newPatchSets = append(newPatchSets, newpatchset)
+	}
+
+	return newPatchSets
+}
+
+func patchPatches(patches []v1.Patch) []Patch {
+	newPatches := []Patch{}
+
+	for _, patch := range patches {
+		newpatch := Patch{}
+		newpatch.Patch = patch
+
+		if patch.Policy != nil && patch.Policy.MergeOptions != nil {
+			newpatch.Policy = patchPolicy(patch.Policy)
+			newpatch.Patch.Policy = nil
+		}
+
+		newPatches = append(newPatches, newpatch)
+	}
+
+	return newPatches
+}
+
+func patchPolicy(policy *v1.PatchPolicy) *PatchPolicy {
+	if policy.FromFieldPath == nil {
```

Comment:
I don't think `FromFieldPath` would matter at all. The way this is written, the function would return early and not produce any policy result if `FromFieldPath` is `nil` but there are valid `MergeOptions` that we should be processing.

Reply:
yes totally agree, fixed now :)